impl/src/main/java/org/eclipse/parsson/JsonGeneratorImpl.java

```diff
@@ -464,6 +476,19 @@ public JsonGenerator writeKey(String name) {
         return this;
     }
 
+    @Override
+    public JsonGenerator writeKey(JsonGenerator.Key name) {
+        if (currentContext.scope != Scope.IN_OBJECT) {
+            throw new JsonGenerationException(
+                    JsonMessages.GENERATOR_ILLEGAL_METHOD(currentContext.scope));
+        }
+        writeName(name);
+        stack.push(currentContext);
```

Comment:
Could refactor extract these 4 lines as they are the same as the lines from `writeKey(String name)`